kernel/async_channel/src/lib.rs

```diff
@@ -103,11 +103,11 @@ impl From<Error> for core2::io::Error {
 /// 
 /// This channel object is not Send/Sync or cloneable itself;
 /// it can be shared across tasks using an `Arc`.
-struct Channel<T: Send> {
+pub struct Channel<T: Send> {
```

Comment:
rather than making the `Channel` type public, or any of its fields or methods, we should simply add methods to `Sender` and `Receiver` that invoke the private `Channel` methods.

I think the only one we really need is `channel_status()`, so you can add a `Sender::channel_status()` and `Receiver::channel_status()` that just call the existing function within `Channel`.

The reason for this is to achieve proper encapsulation through limited visibility -- we don't necessarily want a caller to be able to access any and all fields/methods within `Channel`, because then they could potentially violate its semantics by changing things in unforeseen ways.